eng/common/scripts/Create-APIReview.ps1

```diff
@@ -312,22 +312,27 @@ if ($PackageName)
 {
     Write-Host "Processing $($PackageName)"
     $result = ProcessPackage -packageName $PackageName
-    $responses[$PackageName] = $result 
+    $responses[$PackageName] = $result
 }
 else
 {
-    # process all packages in the artifact
-    foreach ($artifact in $ArtifactList)
-    {
-        Write-Host "Processing $($artifact.name)"
-        $result = ProcessPackage -packageName $artifact.name
-        $responses[$artifact.name] = $result
+    $propFiles = Get-ChildItem -Path $ConfigFileDir -Recurse -Filter "*.json"
+    $artifactPackages = $ArtifactList | ForEach-Object { $_["name"] }
+
+    foreach($propFile in $propFiles) {
+        $artifact = Get-Content $propFile.FullName -Raw | ConvertFrom-Json -AsHashtable
+        if ($artifactPackages -contains $propFile.Name.Replace(".json", ""))
+        {
+            Write-Host "Processing $($artifact.ArtifactName)"
```

Comment:
Isn't the artifact name the same as the file or are their cases where they are different? If they are the same, we probably can avoid reading the json file. If they aren't the same we probably should use the artifactname in your condition instead of the file name.

Reply:
`$pkgInfo.ArtifactName` == `<package-json-file-name>`, if that follows. I'll add comments for clarity and double check, but I'm fairly certain.